mongo_connector/connector.py

```diff
@@ -270,25 +273,31 @@ def read_oplog_progress(self):
                     (name, util.long_to_bson_ts(timestamp))
                     for name, timestamp in data)
 
+    def create_authed_client(self, address=None, **kwargs):
+        kwargs.update(self.ssl_kwargs)
+        if address is None:
+            address = self.address
+        client = MongoClient(address, tz_aware=self.tz_aware, **kwargs)
+        if self.auth_key is not None:
+            client['admin'].authenticate(self.auth_username, self.auth_key)
+        return client
```

Comment:
Refactored `MongoClient` creation and auth so that the new test can easily re-create  `self.main_conn`.